msgraph/applications.go

```diff
@@ -763,3 +763,132 @@ func (c *ApplicationsClient) DeleteFederatedIdentityCredential(ctx context.Conte
 
 	return status, nil
 }
+
+// AssignTokenIssuancePolicy assigns a tokenIssuancePolicy to an application
+func (c *ApplicationsClient) AssignTokenIssuancePolicy(ctx context.Context, application *Application) (int, error) {
```

Comment:
To make this more consistent with other child objects, could we make change this to accept an `applicationId` and a `TokenIssuancePolicy` (e.g. similar to `CreateFederatedIdentityCredential` or `AddPassword`)?

Reply:
Thanks for the review.
Which of the following interfaces do you prefer?
I am willing to adapt to your ideas.

```go
// pattern 1
AssignTokenIssuancePolicy(ctx context.Context, applicationId string, policy TokenIssuancePolicy)
```

```go
// pattern 2
AssignTokenIssuancePolicy(ctx context.Context,  applicationId string, policies *[]TokenIssuancePolicy)
```

Reply:
@manicminer oops. I forgot to mention.

Reply:
@kenchan0130 Apologies for delayed reply, the second form should work well I think?

Reply:
@manicminer 
I don't even know why I wrote codes assigning  TokenIssuancePolicy to the application.
TokenIssuancePolicy should have been tied to the service principal, not the application.
Therefore, I have changed like https://github.com/manicminer/hamilton/pull/215/commits/bbf153dac0f9a28795b4bfda433224f11fe68964.